geomesa-datastore-bundle/geomesa-datastore-processors/src/main/scala/org/geomesa/nifi/datastore/processor/AvroIngestProcessor.scala

```diff
@@ -161,14 +174,14 @@ trait AvroIngestProcessor extends DataStoreIngestProcessor {
         None
       } else if (SimpleFeatureTypes.compare(sft, existing) == 0) {
         None
-      } else if (mode == CompatibilityMode.Update) {
+      } else if (mode == CompatibilityMode.Update && shouldUpdate) {
```

Comment:
i think so yes, although it won't back-fill existing data

Reply:
so basically if you configure the sft explicitly, it will use that to update the existing feature type (if you enable that), and then it will map the input avro file to that feature type. if you don't configure it explicitly, it will use the avro file schema to update the existing feature type (again, if you enable that)

Reply:
that's the best compromise i could come up with to allow for no configuration to work, but also allow for explicit feature types to be used.